tutorials/snapcraft/create-your-first-snap.md

```diff
@@ -0,0 +1,726 @@
+
+
+### Describing the snap
+
+Let’s take a look at the top part of your snapcraft.yaml file. It should look somewhat as shown below:
+
+
+```bash
+name: my-snap-name # you probably want to 'snapcraft register <name>'
+version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
+summary: Single-line elevator pitch for your amazing snap # 79 char long summary
+description: |
+  This is my-snap's description. You have a paragraph or two to tell the
+  most important story about your snap. Keep it under 100 words though,
+  we live in tweetspace and your description wants to look good in the snap
+  store.
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: devmode # use 'strict' once you have the right plugs and slots
+```
+
+This part of `snapcraft.yaml` is mandatory and describes the very basics of the snap metadata. Let’s go through this line by line:
+
+  - **name** describes the name of the snap.
+  - **version** is the current version of the snap. This is just a human readable string. The ascii order doesn’t matter: all snap uploads will get an incremental snap **revision**, which is independant from the version. It’s separated so that you can upload multiple times the same snap for the same architecture with the exact same version. See it as a string that indicates to your user the current version, like “stable”, “2.0” and such.
+  - **summary** is a short, one-line summary or tag-line for your snap.
+  - **description** should provide the user with enough information to judge if the snap is going to be useful to them. This description can span over multiple lines if prefixed with **|**.
+  - **grade** can be used by the publisher to indicate the quality confidence in the build. The store will prevent publishing "devel" grade builds to stable channels.
+  - **confinement** can be either `devmode` or `strict`. We’ll get to this bit later on again. When starting out with a new snap, it’s always a good idea to leave it in `devmode`.
+
+Ok. So much for the basics. Let’s customise this for your very own snap. Please change the top of the file to be:
+
+
+```bash
+name: hello
+version: '2.10'
+summary: GNU Hello, the "hello world" snap
+description: |
+  GNU hello prints a friendly greeting.
+grade: devel
+confinement: devmode
+```
+
+
+```bash
+A quick note on the **version** bit: it’s put into quotation marks ('`2.10`') here, because the version is a string, not a number. You could theoretically use a version string without any numbers (like '`myfirstversion`') and that would be fine too. This version is only for user consumption and doesn’t require anything like ascii ordering (e.g ver1 > ver2 and the like) for an update to reach the user.
+```
```

Comment:
This is an admonition in the original:

![image](https://user-images.githubusercontent.com/7047542/27921995-7b3cf5a4-6271-11e7-89c1-e4b888c3943c.png)